hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -107,21 +109,55 @@
 import static com.hazelcast.util.FutureUtil.logAllExceptions;
 import static com.hazelcast.util.IterableUtil.nullToEmpty;
 import static com.hazelcast.util.Preconditions.checkNotNull;
+import static java.lang.Math.ceil;
+import static java.lang.Math.log10;
 import static java.lang.Math.min;
+import static java.lang.Math.round;
 import static java.util.Collections.singleton;
 import static java.util.concurrent.TimeUnit.SECONDS;
 import static java.util.logging.Level.WARNING;
 
 abstract class MapProxySupport extends AbstractDistributedObject<MapService> implements InitializingObject {
 
     protected static final String NULL_KEY_IS_NOT_ALLOWED = "Null key is not allowed!";
-
     protected static final String NULL_VALUE_IS_NOT_ALLOWED = "Null value is not allowed!";
     protected static final String NULL_PREDICATE_IS_NOT_ALLOWED = "Predicate should not be null!";
     protected static final String NULL_LISTENER_IS_NOT_ALLOWED = "Null listener is not allowed!";
 
     private static final int CHECK_IF_LOADED_TIMEOUT_SECONDS = 60;
 
+    /**
+     * Defines the batch size for operations of {@link IMap#putAll(Map)} calls.
+     *
+     * A value of {@code 0} disables the batching and will send a single operation per member with all map entries.
+     *
+     * If you set this value too high, you may ran into OOME or blocked network pipelines due to huge operations.
+     * If you set this value too low, you will lower the performance of the putAll() operation.
+     */
+    private static final HazelcastProperty MAP_PUT_ALL_BATCH_SIZE
+            = new HazelcastProperty("hazelcast.map.put.all.batch.size", 0);
```

Comment:
would it not make sense to have some other default than 0? Now a user could run into a fat packet and he probably would not quickly figure out what is happening.